apps/prairielearn/src/sync/sharing.ts

```diff
@@ -132,7 +136,15 @@ export function checkInvalidSharingSetAdditions(
         .join(', ')}`,
     );
   }
-  return existInvalidSharingSetAdditions;
+
+  const existDraftQuestionsWithSharingSets = draftQuestionsWithSharingSets.length > 0;
+  if (existDraftQuestionsWithSharingSets) {
+    logger.error(
+      `✖ Course sync completely failed. The following questions are being added to sharing sets, but are drafts: ${draftQuestionsWithSharingSets.join(', ')}`,
+    );
+  }
```

Comment:
Does this handle `sharePublicly` or `shareSourcePublicly`? I'm inclined to say it doesn't.

Possibly language tweak (tagging in @SethPoulsen for opinions):

```suggestion
  const existDraftQuestionsWithSharingSets = draftQuestionsWithSharingSets.length > 0;
  if (existDraftQuestionsWithSharingSets) {
    logger.error(
      `✖ Course sync completely failed. The following draft questions cannot be added to sharing sets: ${draftQuestionsWithSharingSets.join(', ')}`,
    );
  }
```

Reply:
@nwalters512 the language tweak looks good. 

And yes, we should also throw a similar error if `sharePublicly` or `shareSourcePublicly` are true. (Need to also check `sharedPublicly` as well until we finish depracating that).

Reply:
Changed in 562ccc7ca4238b061cc5fed79757979f8c0db35b.